File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/security/AccountUserDetails.java
package net.nemerosa.ontrack.model.security;

import lombok.Data;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.AuthorityUtils;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;

@Data
public class AccountUserDetails implements UserDetails, AccountHolder {

    private final Account account;

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return AuthorityUtils.createAuthorityList(account.getRole().getRoleName());
    }

    @Override
    public String getPassword() {
        return null;
    }

    @Override
    public String getUsername() {
        return account.getName();
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }
}

File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/security/AccountUserDetails.kt
package net.nemerosa.ontrack.model.security;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.AuthorityUtils;
import org.springframework.security.core.userdetails.UserDetails;

 class AccountUserDetails(
         private val account: Account
 ) : UserDetails, AccountHolder {

     override fun getAuthorities(): Collection<GrantedAuthority> =
             AuthorityUtils.createAuthorityList(account.role.roleName)

     override fun isEnabled(): Boolean = true

     override fun getUsername(): String = account.name

     override fun isCredentialsNonExpired(): Boolean {
         TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
     }

     override fun getPassword(): String? = null

     override fun isAccountNonExpired(): Boolean = true

     override fun isAccountNonLocked(): Boolean =true

     override fun getAccount(): Account = account

}
